### ISSUE TITLE
Bug: README for 18.0.0 release config doesn't seem to work

### ISSUE THREAD

**Issue by mriedem** [eslint-community/eslint-plugin-n#527]
### Environment

Node version: 24.11.1
npm version: 11.6.2
ESLint version: 10.2.1
eslint-plugin-n version: 18.0.0
Operating System: ubuntu 24.04

### What rule do you want to report?

Not a rule problem, but a setup problem with the config.

### Link to Minimal Reproducible Example

n/a

### What did you expect to happen?

Following the README for configuring eslint with the plugin should work.

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

Our previous config in eslint.config.js was doing something like this:

```js
const nodePlugin = require("eslint-plugin-n")
...
module.exports = defineConfig([
  ...
  nodePlugin.configs["flat/recommended-script"],
```

That now fails with `TypeError: Cannot read properties of undefined (reading 'flat/recommended-script')`.

Following the README now which says:

```js
import node from "eslint-plugin-n"
import {defineConfig} from "eslint/config"

export default defineConfig([
    plugins: {n: node},
    extends: ["n/recommended-module"],
])
```

First, that looks like a typo since `plugins` and `extends` are properties in an object within an array, right? But there is no object there.

Next, when we put those in an object like this, we get an error:

```js
  {
    plugins: {n: nodePlugin},
    extends: ["n/recommended-module"],
  },
```

>TypeError: Plugin config "recommended-module" not found in plugin "n".

**Comment by aladdin-add** [comment]
thanks for the report! can you re-try as the updated docs: https://github.com/eslint-community/eslint-plugin-n/commit/c1b1b8427e94f751b4e4ffa9d77cbe2d6b4412c9?

**Comment by mriedem** [comment]
> thanks for the report! can you re-try as the updated docs: [c1b1b84](https://github.com/eslint-community/eslint-plugin-n/commit/c1b1b8427e94f751b4e4ffa9d77cbe2d6b4412c9)?

Still getting this error as mentioned above after fixing the object issue:

```
$ npm run lint

> multi-channel-scheduler@1.124.3 lint
> eslint .


Oops! Something went wrong! :(

ESLint: 10.3.0

TypeError: Plugin config "recommended-module" not found in plugin "n".
    at findPluginConfig (/home/osboxes/ibmq/multi-channel-scheduler/node_modules/@eslint/config-helpers/dist/cjs/index.cjs:297:8)
    at /home/osboxes/ibmq/multi-channel-scheduler/node_modules/@eslint/config-helpers/dist/cjs/index.cjs:429:25
    at Array.map (<anonymous>)
    at processExtends (/home/osboxes/ibmq/multi-channel-scheduler/node_modules/@eslint/config-helpers/dist/cjs/index.cjs:427:36)
    at /home/osboxes/ibmq/multi-channel-scheduler/node_modules/@eslint/config-helpers/dist/cjs/index.cjs:503:38
    at Array.flatMap (<anonymous>)
    at processConfigList (/home/osboxes/ibmq/multi-channel-scheduler/node_modules/@eslint/config-helpers/dist/cjs/index.cjs:503:20)
    at defineConfig (/home/osboxes/ibmq/multi-channel-scheduler/node_modules/@eslint/config-helpers/dist/cjs/index.cjs:539:9)
    at Object.<anonymous> (/home/osboxes/ibmq/multi-channel-scheduler/eslint.config.js:7:18)
    at Module._compile (node:internal/modules/cjs/loader:1761:14)
```

This is the full eslint.config.js if that helps:

```
/* eslint-disable n/no-unpublished-require */
const { defineConfig, globalIgnores } = require("eslint/config")
const jest = require("eslint-plugin-jest")
const nodePlugin = require("eslint-plugin-n")
/* eslint-enable n/no-unpublished-require */

module.exports = defineConfig([
  globalIgnores([
    "**/docs",
    // This is locally built docs
    "**/site",
    // The db-migrate generated js template files don't conform to our lint rules.
    "migrations/*.js",
    // These k6 performance tests are no longer maintained and should be removed...
    "tests/performance/*.js"
  ]),
  {
        plugins: {n: nodePlugin},
        extends: ["n/recommended-module"],
  },
  {
    files: ['**/*.test.js'],
    ...jest.configs['flat/recommended']
  },
  {
    files: ['**/*.test.js'],
    ...jest.configs['flat/style']
  },
  {
    plugins: { jest }, // needed to override jest/expect-expect
	rules: {
      "no-process-exit": "off",
      "max-len": ["error", 120],
      "jest/expect-expect": [
        "error",
        { assertFunctionNames: ["expect*"] }
      ]
    }
  },
  {
    files: ["**/*.test.js"],
    rules: {
      // This is needed because we use @quantum/wait-for, jsonwebtoken, supertest, and nock.
      // https://github.com/eslint-community/eslint-plugin-n/blob/HEAD/docs/rules/no-unpublished-import.md
      "n/no-unpublished-require": "off"
    }
  }
])
```

**Comment by aladdin-add** [comment]
emm.. I was not able to repro. This config does indeed exist.: https://github.com/eslint-community/eslint-plugin-n/blob/c1b1b8427e94f751b4e4ffa9d77cbe2d6b4412c9/lib/index.js#L33-L38

what's the eslint version are you using? can you please provide a repro so I can take a look ( something like [playground](https://eslint-online-playground.netlify.app/#eNptjzFuwzAMRa8icGqB2NndtbcoOxgybaiVSUGSgxSG717KUoAMWSiReJ//c4cU7ZXu4xo89T8JBrDCSbTxsryFKJZS6olv7x/IcAFK3nHuFZrdUgVuDRKzYTNHWQ02pAt+Wxx3jKBKZLqf1ETzuPlsvpCN2UsxppJpMLsuOS51GDdPZVQ7o3v5ytK1RJ0mQhiKW4wSESp2lEfLdw0bRvs7LuUuYQ167kLIf4GqdpVJXRBOS4SJbp8UiCdi6ygVptk/jqoyP2ZK+m9JX1z8TLVIBxz/MqF45g==))?

**Comment by aladdin-add** [comment]
I see. if using cjs, you should be using `.default` (the v18 is esm-only):

```js
const nodePlugin = require("eslint-plugin-n").default;
```

**Comment by mriedem** [comment]
> I see. if using cjs, you should be using `.default` (the v18 is esm-only):
> 
> const nodePlugin = require("eslint-plugin-n").default;

Yeah this works, thank you!
